src/mlpack/core/dists/gaussian_distribution.hpp

```diff
@@ -103,7 +103,24 @@ class GaussianDistribution
    * @param logProbabilities Output log probabilities for each input
    *     observation.
    */
-  void LogProbability(const arma::mat& x, arma::vec& logProbabilities) const;
+  void LogProbability(const arma::mat& x, arma::vec& logProbabilities) const
+  {
+    // Column i of 'diffs' is the difference between x.col(i) and the mean.
+    arma::mat diffs = x - (mean * arma::ones<arma::rowvec>(x.n_cols));
+
+    // Now, we only want to calculate the diagonal elements of (diffs' * cov^-1
+    // * diffs).  We just don't need any of the other elements.  We can
+    // calculate the right hand part of the equation (instead of the left side)
+    // so that later we are referencing columns, not rows -- that is faster.
+    const arma::mat rhs = -0.5 * invCov * diffs;
+    arma::vec logExponents(diffs.n_cols); // We will now fill this.
+    for (size_t i = 0; i < diffs.n_cols; i++)
+      logExponents(i) = accu(diffs.unsafe_col(i) % rhs.unsafe_col(i));
+
+    const size_t k = x.n_rows;
```

Comment:
Hm, why not just use ``x.n_rows`` instead of ``k``, it's only used once.

Reply:
I agree with this. I think we should remove this. Thanks